packages/rolldown/src/rolldown-build.ts

```diff
@@ -43,17 +39,15 @@ export class RolldownBuild {
   }
 
   async close(): Promise<void> {
-    const bundler = await this.#getBundler({})
-    await this.#stopWorkers?.()
-    await bundler.close()
-  }
-
-  async watch(outputOptions: OutputOptions = {}): Promise<Watcher> {
-    const bundler = await this.#getBundler(outputOptions)
-    const bindingWatcher = await bundler.watch()
-    const watcher = new Watcher(bindingWatcher)
-    watcher.watch()
-    return watcher
+    // If the bundler not create, create one to make close related things(eg. hooks) could be work.
+    if (this.#bundlers.length === 0) {
+      await this.#getBundler({})
```

Comment:
why not just return?

Reply:
It will break the behavior

``` .tsx
const bundle = await rollup({
        input: './main.js',
        plugins: [
            {
              closeBundle: () => {
                console.log(1111) // should called
              },
            },
          ],
    });
    await bundle.close()
```